### PR TITLE
show-error-label-when-server-is-unresponsive

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**
image

### CODE REVIEW

File: app/scripts/proactive/view/LogoutView.js [hunk 1]
```diff
@@ -52,6 +52,11 @@ define(
                 function tryToConnect() {
                     if (StudioClient.isLoggedIn()) {
                         isConnected()
+                    } else {
+                        var connectionError = $('<div id="connection-error" class="btn-group pa-tools"><label style="color:red; font-weight:bold; font-size:15px; padding-left:4px; padding-top:4px;">You have been disconnected from the server. Reload the page in order to log into the portal again. </label></div>');
```
Comment (medou-boushab) [inline app/scripts/proactive/view/LogoutView.js:56]: why adding this in logout view ?

Comment (oanaBiancaSchiopu) [reply app/scripts/proactive/view/LogoutView.js:56]: this is where the check is done if the user is still logged in:
 if (StudioClient.isLoggedIn()) {

File: app/scripts/proactive/rest/studio-client.js [hunk 2]
```diff
@@ -197,7 +197,11 @@ define(
                             }
                         },
                         error: function(response) {
-                            console.error("Unknown User Permission", response)
+                            console.error("Unknown User Permission", response);
+                            var connectionError = $('<div id="connection-error" class="btn-group pa-tools"><label style="color:red; font-weight:bold; font-size:15px; padding-left:4px; padding-top:4px;">The server cannot be reached. Check the server status and your network connection. </label></div>');
```
Comment (medou-boushab) [inline app/scripts/proactive/rest/studio-client.js:201]: I think it's not the right place, you should move it to the ligne 183

File: app/scripts/proactive/view/LogoutView.js (hunk 1 above)
Comment (medou-boushab) [reply app/scripts/proactive/view/LogoutView.js:56]: yes, this means that you will wait for the second check:
isLoggedIn returns localStorage['pa.session'] != null
so before removing the session id it doesn't mean you are still connected.

File: app/scripts/proactive/rest/studio-client.js (hunk 2 above)
Comment (oanaBiancaSchiopu) [reply app/scripts/proactive/rest/studio-client.js:201]: if I put it to line 183, it won't be executed in the case when the server is not running

File: app/scripts/proactive/view/LogoutView.js (hunk 1 above)
Comment (oanaBiancaSchiopu) [reply app/scripts/proactive/view/LogoutView.js:56]: where do you think this should be added?

**Review by medou-boushab — APPROVED** [review]

**Pull request merged**